tests/test_modules/test_pmac/test_motorinfo.py

```diff
@@ -18,16 +20,49 @@ def setUp(self):
             units="mm"
         )
 
+    def check_distance(self, d, times, velocities):
+        # sum the area of two trapezoids z1, z2 and a rectangle r
+        z1 = (velocities[0] + velocities[1]) / 2 * times[1]
+        st = self.o.velocity_settle
+        if (len(times) == 3 and st == 0) or (len(times) == 4 and st > 0):
+            r = 0
+            z2 = (velocities[1] + velocities[2]) / 2 * (times[2] - times[1])
+        else:
+            r = velocities[1] * (times[2] - times[1])
+            z2 = (velocities[2] + velocities[3]) / 2 * (times[3] - times[2])
+        sd = velocities[-1] * (times[-1] - times[-2]) if st > 0 else 0
+        assert np.isclose(z1 + z2 + r + sd, d)
+
     def test_turnaround(self):
         # 0_| \
         #   |  \
         v1 = 0.1
         v2 = -0.1
         distance = 0.0
         time_array, velocity_array = self.o.make_velocity_profile(
+            v1, v2, distance, 0.0).make_arrays()
+        self.check_distance(distance, time_array, velocity_array)
+        assert np.isclose(time_array, [0.0, 0.05, 0.1]).all()
```

Comment:
```suggestion
        assert time_array == pytest.approx([0.0, 0.05, 0.1])
```

Maybe a bit more readable? Debatable though...

Reply:
it looks a bit nicer

Reply:
Done